model/shared/src/main/scala/explore/model/ScienceModeBasic.scala

```diff
@@ -16,27 +17,18 @@ import lucuma.core.enums.GmosSouthGrating
 import monocle.Focus
 import monocle.Lens
 
-sealed trait ScienceModeBasic extends Product with Serializable
+sealed trait ScienceModeBasic extends Product with Serializable derives Eq
 
 object ScienceModeBasic {
-  implicit val scienceModeBasicEq: Eq[ScienceModeBasic] =
-    Eq.instance {
-      case (a: GmosNorthLongSlit, b: GmosNorthLongSlit) => a === b
-      case (a: GmosSouthLongSlit, b: GmosSouthLongSlit) => a === b
-      case _                                            => false
-    }
-
   final case class GmosNorthLongSlit(
     grating: GmosNorthGrating,
     filter:  Option[GmosNorthFilter],
     fpu:     GmosNorthFpu
   ) extends ScienceModeBasic
+      derives Eq
```

Comment:
I'm wondering if this `derives Eq` is necessary or the one in the trait derives anyway and kicks in when we compare subtypes.

Reply:
Good question. I guess it depends on whether you ever call `===` on the subtype

Reply:
They are used to define `Reusaibility` somewhere